app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,97 @@
+
+
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: task_id)
+
+
+    if @task.nil?
+      head 302
+      return
+    end
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(
+        name: params[:task][:name],
+        description: params[:task][:description],
+        completed_at: params[:task][:completed_at]
+    )
+    if @task.save
+      redirect_to task_path(@task.id)
+      return
+    else
+      render :new
+      return
+    end
+  end
+
+  def edit
+    @task = Task.find_by(id: params[:id])
+
+    if @task.nil?
+      redirect_to tasks_path
+      return
+    end
+  end
+
+  def update
+    @task = Task.find_by(id: params[:id])
+    task_params = params.require(:task).permit(:name, :description, :completed_at)
+    if @task.nil?
+      redirect_to root_path
+      return
+    elsif @task.update(task_params)
+      redirect_to task_path(@task.id)
+      return
+    else
+      render :edit
+      return
+    end
+  end
+
+  def destroy
+    @task = Task.find_by(id: params[:id])
+    task_params = params.require(:task).permit(:name, :description, :completed_at)
```

Comment:
Once you fix the above, you will see the following error: 
```
ParameterMissing: param is missing or the value is empty: task
```
This is because none of these params are given during a delete request. Only the id is given, so this line is unnecessary.
```suggestion
```